src/solveh.rs

```diff
@@ -252,12 +252,9 @@ where
 }
 
 /// An interface for calculating determinants of Hermitian (or real symmetric) matrix refs.
-pub trait DeterminantH {
-    type Output;
-    type SignLnOutput;
-
+pub trait DeterminantH<A: Scalar> {
```

Comment:
Type parameter `A: Scalar` should be an associated type:

```rust
pub trait DeterminantH {
  type Output: Scalar;
  ...
} 
```

Type parameter for trait means it can be implemented twice for a struct, e.g.

```rust
impl DeterminantH<A> for ArrayBase<S, Ix2> { .. }
impl DeterminantH<Result<A>> for ArrayBase<S, Ix2> { ... }
```

This approach is indeed possible for this case, but we do not choose it on PR. We should use associated types instead of type parameter for traits as possible.

Reply:
That makes sense. Is it okay if I call the associated type `Elem` instead of `Output`, though? The associated type is the type of elements in the matrix, not the output type of the methods. (I suppose it could be `type OutElem = A::Real` instead of `type Elem = A`, but I think using `A` is clearer.)

Reply:
yes, `Elem` is better